app/screens/LoginScreen/loginScreen.js

```diff
@@ -9,132 +9,103 @@ import {
     } else {
-      alert("Please enter A Valid Email");
+      this.login();
     }
-  };
-
-  onPressLogin() {
-    LoginManager.logInWithReadPermissions(["public_profile", "email"]).then(
-      result => this._handleCallBack(result),
-      function (error) {
-        alert("Login fail with error: " + error);
-      }
-    );
   }
 
-  _handleCallBack(result) {
-    let _this = this;
-    if (result.isCancelled) {
-      alert("Login cancelled");
-    } else {
-      AccessToken.getCurrentAccessToken().then(data => {
-        const token = data.accessToken;
-        fetch(
-          "https://graph.facebook.com/v2.8/me?fields=id,first_name,last_name,gender,birthday&access_token=" +
-            token
-        )
-          .then(response => response.json())
-          .then(json => {
-            const imageSize = 120;
-            const facebookID = json.id;
-            const fbImage = `https://graph.facebook.com/${facebookID}/picture?height=${imageSize}`;
-            this.authenticate(data.accessToken).then(function (result) {
-              const { uid } = result;
-              _this.createUser(uid, json, token, fbImage);
-            });
-          })
-          .catch(function (err) {
-            console.log(err);
-          });
-      });
+  // Signin with email and password
+  async login() {
+    try {
+      let email = this.state.email;
+      let password = this.state.password;
+
+      const res = await signInWithEmailAndPassword(auth, email, password);
+    } catch (error) {
+      alert(error.message.toString());
     }
   }
 
-  authenticate = token => {
-    const provider = auth.FacebookAuthProvider;
-    const credential = provider.credential(token);
-    let ret = auth.signInWithCredential(credential);
-    return ret;
-  };
-
-  createUser = (uid, userData, token, dp) => {
-    const defaults = {
-      uid,
-      token,
-      dp,
-      ageRange: [20, 30],
-    };
-    f.database()
-      .ref("users")
-      .child(uid)
-      .update({ ...userData, ...defaults });
-  };
-
-  _signInAsync = async () => {
-    if (EmailValidator.validate(this.state.email) === true) {
-      if (this.state.Pasword != "") {
-        this.login();
-      } else {
-        alert("Enter the password");
+  // Signin user with facebook account
+  async _fbSignInAsync() {
+    try {
+      const result = await LoginManager.logInWithPermissions(["public_profile", "email"]);
+      if (!result.isCancelled) {
+        // Get the user's access token
+        const data = await AccessToken.getCurrentAccessToken();
+
+        // Use the user's access token to authenticate with Firebase Authentication
+        const credential = FacebookAuthProvider.credential(data.accessToken);
+        const { user } = await signInWithCredential(auth, credential);
+
+        const response = await fetch(
+          `https://graph.facebook.com/v2.8/me?fields=id,first_name,last_name,email,birthday&access_token=${data.accessToken}`
```

Comment:
I got this on internet 

"Uses of **await** or **Promise.allSettled()** depends on the specific use case and the performance characteristics of the promises being used. In general, if the promises can execute in parallel and the outcome of rejected promises is not important, Promise.allSettled() can be a good choice. However, if the promises need to execute sequentially or the outcome of rejected promises needs to be handled specifically, await may be a better choice."

Also in our case the response of promises are dependent on each other.

Reply:
@shehand Whats your suggestion on it.

Reply:
Yes you are right, these two promises are independent of each other but they are dependent on the above promises, which are not visible in this snippet.

So, I can either execute these two promises in parallel using Promise.allSettled() and explicitly check for their success and failure (since Promise.allSettled() will wait for all functions passed in the array to either resolve or reject without rejecting itself), 
or I can use Promise.all(), which will reject as soon as one of the functions passed in the array rejects.